### PR TITLE
Improve shell scripts in our Docker images

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Bugfix

### Description

We seem to have fairly often problems with the quality of the shell scripts in our container images where for example errors when preparing TLS certificates are ignored and do not fail the container startup (latest in #3092).

This PR tries to improve it by setting `-e` in all the scripts to make them fail on error. Additionally, it also adds validation using [Shellcheck](https://www.shellcheck.net/) which is also integrated into the CI build and resulted in some more changes to actually pass the checks.

### Checklist

- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging

### CODE REVIEW

File: user-operator/scripts/user_operator_run.sh [hunk 1]
```diff
@@ -14,4 +16,4 @@ fi
 
 export JAVA_CLASSPATH=lib/io.strimzi.@project.build.finalName@.@project.packaging@:@project.dist.classpath@
 export JAVA_MAIN=io.strimzi.operator.user.Main
-exec ${STRIMZI_HOME}/bin/launch_java.sh
+exec "${STRIMZI_HOME}"/bin/launch_java.sh
```
Comment (samuel-hawker) [inline user-operator/scripts/user_operator_run.sh:19]: Normally you would quote the whole string I.e. exec "${STRIMZI_HOME}/bin/launch_java.sh"

I think in practice is has little difference but it makes it a little clearer to read

File: docker-images/jmxtrans/docker-entrypoint.sh [hunk 2]
```diff
@@ -17,8 +16,10 @@ MONITOR_OPTS="-Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.ssl=
               -Djava.rmi.server.hostname=${PROXY_HOST}"
 
 if [ "$1" = 'start-without-jmx' ]; then
+    # shellcheck disable=SC2086
     set /usr/bin/tini -w -e 143 -- java -server $JAVA_OPTS $JMXTRANS_OPTS $GC_OPTS $EXEC
```
Comment (samuel-hawker) [inline docker-images/jmxtrans/docker-entrypoint.sh:20]: I think you can remove the sc disable line if you write:
```
set /usr/bin/tini -w -e 143 -- "java -server ${JAVA_OPTS} ${JMXTRANS_OPTS} ${GC_OPTS} ${EXEC}"
```
Might be wrong on this.

File: docker-images/kafka/exporter-scripts/kafka_exporter_run.sh [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
+set -e
 set +x
 
 # We don't need LOG_DIR because we write no log files, but setting it to a
 # directory avoids trying to create it (and logging a permission denied error)
 export LOG_DIR="$KAFKA_EXPORTER_HOME"
 
 if [ -n "$KAFKA_EXPORTER_GROUP_REGEX" ]; then
+    # shellcheck disable=SC2027
     groupregex="--group.filter=\""${KAFKA_EXPORTER_GROUP_REGEX}"\""
```
Comment (samuel-hawker) [inline docker-images/kafka/exporter-scripts/kafka_exporter_run.sh:11]:
```
groupregex="--group.filter=\"${KAFKA_EXPORTER_GROUP_REGEX}\""
```
Should mean we can remove the disable

Comment (scholzj) [reply docker-images/kafka/exporter-scripts/kafka_exporter_run.sh:11]: I'm not completely sure about it - I think removing it breaks the functionality with some settings.

File: docker-images/jmxtrans/docker-entrypoint.sh (hunk 2 above)
Comment (scholzj) [reply docker-images/jmxtrans/docker-entrypoint.sh:20]: I gave this a try and it doesn't work. Tini takes it and tries to execute it as single executable - and says that such file/directory does not exist.

**Comment by scholzj** [comment]
@strimzi/maintainers Ok, the tests seem to be passing on this now. So this should be ready for review. Please have a look.

**Comment by scholzj** [comment]
@samuel-hawker Yeah, it is a bit unfortunate. I think we should ideally continue and make the other scripts compliant as well. That way we cn hopefully at. the end run it on everything. But there is lot of scripts to work on - I didn't want to make the PR too big, so I focused on thos in the Docker images as they are most important.

**Pull request merged**